server/queue/queue-pulsar/src/test/java/org/apache/james/queue/pulsar/PulsarMailQueueTest.java

```diff
@@ -316,4 +324,58 @@ public void browseShouldReturnEmptyWhenSingleDequeueMessageEvenWhenStoreIsGuaran
                 .isEmpty();
     }
 
+    @Test
+    void dequeueShouldAllowToNackAndRetrieveSeveralMails() throws Exception {
+        enQueue(defaultMail()
+                .name("name1")
+                .build());
+        enQueue(defaultMail()
+                .name("name2")
+                .build());
+        enQueue(defaultMail()
+                .name("name3")
+                .build());
+
+        Iterator<MailQueue.MailQueueItem> items = Flux.from(getMailQueue().deQueue()).subscribeOn(SCHEDULER).toIterable().iterator();
+        MailQueue.MailQueueItem item1 = items.next();
+        log.info("mail to retry {}", item1.getMail().getName());
+        item1.done(MailQueue.MailQueueItem.CompletionStatus.RETRY);
+        MailQueue.MailQueueItem item3 = items.next();
+        log.info("mail to process {}", item3.getMail().getName());
+        item3.done(MailQueue.MailQueueItem.CompletionStatus.SUCCESS);
+
+        MailQueue.MailQueueItem item2 = items.next();
+        log.info("mail to retry {}", item2.getMail().getName());
+        item2.done(MailQueue.MailQueueItem.CompletionStatus.RETRY);
+
+        MailQueue.MailQueueItem item4 = items.next();
+        log.info("mail to process {}", item4.getMail().getName());
+        item4.done(MailQueue.MailQueueItem.CompletionStatus.SUCCESS);
+
+        MailQueue.MailQueueItem item5 = items.next();
+        log.info("mail to retry {}", item5.getMail().getName());
+        item5.done(MailQueue.MailQueueItem.CompletionStatus.RETRY);
+
+        MailQueue.MailQueueItem item6 = items.next();
+        log.info("mail to process {}", item6.getMail().getName());
+        item6.done(MailQueue.MailQueueItem.CompletionStatus.SUCCESS);
+
+
+        assertThat(Stream.of(item3, item4, item5).map(item -> item.getMail().getName()))
+                .containsOnly("name1", "name2","name3");
+    }
+
+    @Tag(Unstable.TAG)
+    @Override
+    public void removeByRecipientShouldRemoveSpecificEmail() throws Exception {
+        DelayedManageableMailQueueContract.super.removeByRecipientShouldRemoveSpecificEmail();
+    }
+
+    @ParameterizedTest
```

Comment:
this test is missing the unstable tag but we will actually propose a proper fix for flakyness